migration_original/ODS1Stage/tables/Mid/ProviderSponsorship/spu_translated_ProviderSponsorship.sql

```diff
@@ -0,0 +1,1544 @@
+where
+    b.entitytypecode = 'CLPROD'
+    and c.messagetypecode = 'SAFEHARBOR'
+    and displayonpage = 1
+    and d.isactive = 1
+    and e.messagepagecode = 'SPONSPHYPRO'
+), 
+cte_entity_message_state AS (
+select
+    cetcf.entityid as clienttoproductid,
+    cf.clientfeaturecode
+from
+    base.cliententitytoclientfeature cetcf
+    join base.entitytype et on cetcf.entitytypeid = et.entitytypeid
+    join base.clientfeaturetoclientfeaturevalue cftcfv on cetcf.clientfeaturetoclientfeaturevalueid = cftcfv.clientfeaturetoclientfeaturevalueid
+    join base.clientfeature cf on cftcfv.clientfeatureid = cf.clientfeatureid
+    join base.clientfeaturevalue cfv on cfv.clientfeaturevalueid = cftcfv.clientfeaturevalueid
+where
+    et.entitytypecode = 'CLPROD'
+    and cf.clientfeaturecode = 'FCSBS' -- State Border for Sponsorship
+    and cfv.clientfeaturevaluecode = 'FVYES'
+),
+cte_practice as (
+select
+    cper.parentid,
+    off.officeid,
+    off.officecode,
+    off.officename,
+    prac.practiceid,
+    prac.practicecode,
+    prac.practicename,
+    cpte.clientproducttoentityid
+from
+    base.clientproductentityrelationship cper
+    join base.relationshiptype rt on cper.relationshiptypeid = rt.relationshiptypeid
+    join base.clientproducttoentity cpte on cpte.clientproducttoentityid = cper.childid
+    join base.office off on cpte.entityid = off.officeid
+    left join base.practice prac on off.practiceid = prac.practiceid
+where
+    rt.relationshiptypecode = 'PROVTOOFF'
+),
+
+cte_clientcode_in as (
+select distinct
+    cl.clientcode
+from
+    base.cliententitytoclientfeature cetcf
+    join base.entitytype et on cetcf.entitytypeid = et.entitytypeid
+    join base.clientfeaturetoclientfeaturevalue cftcfv on cetcf.clientfeaturetoclientfeaturevalueid = cftcfv.clientfeaturetoclientfeaturevalueid
+    join base.clientfeature cf on cftcfv.clientfeatureid = cf.clientfeatureid
+    join base.clientfeaturevalue cfv on cfv.clientfeaturevalueid = cftcfv.clientfeaturevalueid
+    join base.clientfeaturegroup cfg on cf.clientfeaturegroupid = cfg.clientfeaturegroupid
+    join base.clienttoproduct ctp on ctp.clienttoproductid = cetcf.entityid
+    join base.client cl on cl.clientid = ctp.clientid
+    join base.product prod on ctp.productid = prod.productid
+    join base.productgroup pg on prod.productgroupid = pg.productgroupid
+where
+    et.entitytypecode = 'CLPROD'
+    and cf.clientfeaturecode = 'FCDOA'
+    and cfv.clientfeaturevaluecode = 'FVYES'
+    and pg.productgroupcode = 'PDC'
+),
+
+cte_pracoffice_update as (
+select
+    pb.providercode,
+    off.officecode,
+    prac.practicecode,
+    pto.providerofficerank,
+    off.officeid,
+    off.officename,
+    prac.practiceid,
+    prac.practicename,
+    off.officecode as officecodetemp,
+    prac.practicecode as practicecodetemp,
+    row_number() over (partition by pb.providercode order by pto.providerofficerank asc, off.officecode asc) as recid
```

Comment:
I see this column is not being used anywhere else, and the function is costly; it could be removed.